### PR TITLE
fix: Prevent invalid session handlers

### LINKED ISSUE #9024: Bug: Class 'file' not found in SYSTEMPATH\Config\Services.php on line 686.
### PHP Version

7.4

### CodeIgniter4 Version

4.4.8

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

SQL Server 15

### What happened?

I get this error when I deploy the application to the server, but on my local machine this never happens.
When the error appears and refresh the browser it returns to normal, but it becomes annoying because it can appear more than 10 times every day.

### Steps to Reproduce

error in log file:

```
Error: Class 'file' not found
in SYSTEMPATH\Config\Services.php on line 686.
 1 SYSTEMPATH\Config\BaseService.php(258): CodeIgniter\Config\Services::session(Object(Config\Session), false)
 2 SYSTEMPATH\Config\BaseService.php(199): CodeIgniter\Config\BaseService::__callStatic('session', [...])
 3 SYSTEMPATH\Config\Services.php(664): CodeIgniter\Config\BaseService::getSharedInstance('session', null)
 4 SYSTEMPATH\Config\BaseService.php(258): CodeIgniter\Config\Services::session()
 5 SYSTEMPATH\Common.php(982): CodeIgniter\Config\BaseService::__callStatic('session', [])
 6 APPPATH\Filters\Dokterpriv\POLI.php(30): session()
 7 SYSTEMPATH\Filters\Filters.php(184): App\Filters\Dokterpriv\POLI->before(Object(CodeIgniter\HTTP\IncomingRequest), null)
 8 SYSTEMPATH\CodeIgniter.php(475): CodeIgniter\Filters\Filters->run('dokter/pasien/get-pasien-list', 'before')
 9 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
10 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
```

I have checked the Sessions.php config file in my app folder and there is nothing wrong. I have followed the existing documentation.

### Expected Output

the session should be able to run as it should.

### Anything else?

I've searched Google and I only got 1 reference :

https://stackoverflow.com/questions/76528607/codeigniter-error-class-file-not-found-systempath-config-services-php-at-lin

and that doesn't solve the problem either.

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Fixes #9024 

Prevent cases where invalid session handler is provided.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by kenjis — APPROVED** [review]

File: system/Config/Services.php [hunk 1]
```diff
@@ -674,17 +676,24 @@ public static function session(?SessionConfig $config = null, bool $getShared =
...
-            if ($driver === 'MySQLi') {
+            if ($driverPlatform === 'MySQLi') {
                 $driverName = MySQLiHandler::class;
-            } elseif ($driver === 'Postgre') {
+            } elseif ($driverPlatform === 'Postgre') {
                 $driverName = PostgreHandler::class;
             }
         }
 
+        if (! class_exists($driverName) || ! is_a($driverName, SessionBaseHandler::class, true)) {
```
Comment (kenjis) [inline system/Config/Services.php:689]: Should be `SessionHandlerInterface`?

File: system/Config/Services.php [hunk 2]
```diff
@@ -674,17 +676,24 @@ public static function session(?SessionConfig $config = null, bool $getShared =
...
+            throw new InvalidArgumentException(sprintf(
+                'Invalid session handler "%s" provided.',
+                $driverName
+            ));
+        }
+
+        /** @var SessionBaseHandler $driver */
```
Comment (kenjis) [inline system/Config/Services.php:696]: Should be `SessionHandlerInterface`?

File: system/Config/Services.php (hunk 1 above)
Comment (paulbalandan) [reply system/Config/Services.php:689]: SessionHandlerInterface does not have the setLogger() method used in the following lines.

Comment (kenjis) [reply system/Config/Services.php:689]: Okay.

**Review by michalsn — APPROVED** [review]

**Pull request merged**